mypy/modulefinder.py

```diff
@@ -805,15 +810,29 @@ def compute_search_paths(sources: List[BuildSource],
                        typeshed_path=tuple(lib_path))
 
 
-def load_stdlib_py_versions(custom_typeshed_dir: Optional[str]) -> StdlibVersions:
+def get_typeshed_dir(python_executable: Optional[str] = None) -> str:
+    packages_dirs = get_site_packages_dirs(python_executable)[1]
+    typeshed_dir_name = "basedtypeshed"
+    for dir in packages_dirs:
+        result = os.path.join(dir, typeshed_dir_name)
+        if os.path.isdir(result):
+            return result
+    raise StopIteration(
```

Comment:
Why `StopIteration`?

Reply:
because its iterating over `packages_dirs`

what on earth is the industry standard for figuring out what exception to raise

Reply:
https://docs.python.org/3/library/exceptions.html#StopIteration

A `StopIteration` is used to indicate that there is nothing `next`.

Reply:
so what should it raise instead?

Reply:
`FileNotFoundError`?